src/runtime/components/SelectMenu.vue

```diff
@@ -193,6 +199,24 @@ const groups = computed(() => props.items?.length ? (Array.isArray(props.items[0
 // eslint-disable-next-line vue/no-dupe-keys
 const items = computed(() => groups.value.flatMap(group => group) as T[])
 
+const creatable = computed(() => {
+  if (!props.creatable) {
+    return false;
+  }
+
+  const filteredItems = filterFunction(items.value, searchTerm.value)
```

Comment:
this is the slight tricky bit, need to keep track of what the active search shows (and then have an additional option with the custom input)